edgedb-protocol/src/server_message.rs

```diff
@@ -271,7 +268,15 @@ impl ServerMessage {
             0x52 => Authentication::decode(data).map(M::Authentication),
             0x5a => ReadyForCommand::decode(data).map(M::ReadyForCommand),
             0x4b => ServerKeyData::decode(data).map(M::ServerKeyData),
-            0x53 => ParameterStatus::decode(data).map(M::ParameterStatus),
+            0x53 => {
+                if &Bytes::decode(data)?[..] == b"system_config" {
```

Comment:
Why do you need this hack? Maybe just parse it in parameter_status loop?

Reply:
Oh I was trying to minimize the copying but still ended up with copying for readability, so yeah this hack doesn't make sense now.